backend/src/main/java/com/redhat/cloud/notifications/routers/EventLogMigrationService.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.redhat.cloud.notifications.models.Event;
+import com.redhat.cloud.notifications.models.NotificationHistory;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.redhat.cloud.notifications.Constants.INTERNAL;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+// TODO [Event log phase 2] Delete this class after the data migration.
+@Path(INTERNAL + "/event_log/migrate")
+public class EventLogMigrationService {
+
+    public static final String CONFIRMATION_TOKEN = "ready-set-go";
+
+    private static final Logger LOGGER = Logger.getLogger(EventLogMigrationService.class);
+
+    @Inject
+    Mutiny.Session session;
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    public Uni<MigrationReport> migrate(@QueryParam("confirmation-token") String confirmationToken) {
+        long start = System.currentTimeMillis();
+        if (!CONFIRMATION_TOKEN.equals(confirmationToken)) {
+            throw new BadRequestException("You forgot to confirm the migration!");
+        }
+        MigrationReport report = new MigrationReport();
+        // All DB changes will be rolled-back if anything goes wrong.
+        return session.withTransaction(tx -> {
+            LOGGER.debug("[NOTIF-291] Start");
+            return session.createQuery("FROM NotificationHistory WHERE event IS NULL", NotificationHistory.class)
+                    .getResultList()
+                    .onItem().transformToMulti(Multi.createFrom()::iterable)
+                    .onItem().transformToUniAndConcatenate(history -> {
+                        Event event = new Event();
+                        event.setAccountId(history.getAccountId());
+                        return session.persist(event)
```

Comment:
mmh, IIUC this will create one `Event` per each `NotificationHistory`? 

AFAIK one Event can cause multiple `NotificationHistory` (i.e. one event triggering an email and a 2 webhook should create 3 `NotificationHistory`) - but I don't think we had any way to identify `NotificationHistory` create from the same event.

If that's the case, I'm wondering if we should do this migration.

Reply:
You understood correctly, but here's the reason why we need that migration.

After this PR, `NotificationResources#getNotificationHistory` and `NotificationResources#getNotificationDetails` will need a `NotificationHistory <-> Event` relationship to return a non-empty/null result. If we don't migrate the current prod data, the _old_ history will disappear from the UI after we deploy this branch on prod. The migration will prevent that issue by adding fake `Events` but as you sensed it wouldn't make sense to display these `Events` in the UI because they won't reflect what really happened. That's why `Events` won't be displayed for 14 days (the retention delay) in the UI.

Here's the plan I had in mind:
- We deploy this branch to prod ASAP. Fake `Events` are added to the current history to keep everything running as usual with the old UI.
- For 14 days, the history is filled with real `Events` linked to the new `NotificationHistory` entries and the old history data with fake `Events` is slowly removed by a scheduled job that I will submit as a new PR tomorrow.
- After 14 days, the history only contains real `Events` and that's the moment when we can display the new event log UI.

This is very similar to what we did with policies-history.

WDYT?

Reply:
TL;DR The migration result is not meant to be displayed with the new event log UI. Its only goal is to avoid breaking the old UI. 😄

Reply:
Makes sense!